crates/rust-analyzer/src/main_loop.rs

```diff
@@ -197,7 +198,28 @@ impl GlobalState {
                     }
                 }
                 Task::Workspaces(workspaces) => self.switch_workspaces(workspaces),
-                Task::Unit => (),
+                Task::PrimeCaches(progress) => {
+                    let (state, message, percentage);
+                    match progress {
+                        PrimeCachesProgress::Started => {
+                            state = Progress::Begin;
+                            message = None;
+                            percentage = 0.0;
+                        }
+                        PrimeCachesProgress::StartedOnCrate { on_crate, n_done, n_total } => {
+                            state = Progress::Report;
+                            message = Some(format!("{}/{} ({})", n_done, n_total, on_crate));
+                            percentage = Progress::percentage(n_done, n_total);
+                        }
+                        PrimeCachesProgress::Finished => {
+                            state = Progress::End;
+                            message = None;
+                            percentage = 1.0;
```

Comment:
While it doesn't really matter (percentage is not sent over for a progress end) I think this should be `100.0` to be consistent.

Reply:
I think we should fix this the other way around: https://github.com/rust-analyzer/rust-analyzer/pull/6166

(rename to `fraction` is still warranted though)